src/blockdata/block.rs

```diff
@@ -308,12 +309,14 @@ impl fmt::Display for Bip34Error {
     }
 }
 
-impl ::std::error::Error for Bip34Error {}
+#[cfg(feature = "std")]
+impl std::error::Error for Bip34Error {}
```

Comment:
I don't quite understand what a lack of an `Error` impl in `no_std would mean here ...

Reply:
Not too much. The whole Error concept is std-only usually. The `Result` type doesn't require the second generic parameter to be `Error`, so it has very few effects.